.github/workflows/nightly_detect_failure.yml

```diff
@@ -205,7 +205,41 @@ jobs:
           Current content: ${{ steps.bump_version.outputs.toolchain_content }}
           This needs to be fixed for the nightly testing process to work correctly.
 
-    - name: Compare versions and post a reminder.
+    - name: Setup for automatic PR creation
+      if: steps.check_branch.outputs.result == 'false'
+      env:
+        BUMP_VERSION: ${{ steps.bump_version.outputs.result }}
+        BUMP_BRANCH: ${{ steps.latest_bump_branch.outputs.result }}
+        SHA: ${{ env.SHA }}
+      run: |
+        # Install zulip-send for the script
+        pip install zulip-send
+        # Configure git for the bot
+        git config --global user.name "mathlib4-bot"
+        git config --global user.email "github-mathlib4-bot@leanprover.zulipchat.com"
+        # Store credentials for zulip-send
+        echo "ZULIP_EMAIL=github-mathlib4-bot@leanprover.zulipchat.com" >> ~/.zuliprc
+        echo "ZULIP_API_KEY=${{ secrets.ZULIP_API_KEY }}" >> ~/.zuliprc
+        echo "ZULIP_SITE=https://leanprover.zulipchat.com" >> ~/.zuliprc
+        chmod 600 ~/.zuliprc
+
+    - name: Attempt automatic PR creation
+      id: auto_pr
+      if: steps.check_branch.outputs.result == 'false'
+      continue-on-error: true
+      env:
+        BUMP_VERSION: ${{ steps.bump_version.outputs.result }}
+        BUMP_BRANCH: ${{ steps.latest_bump_branch.outputs.result }}
+        SHA: ${{ env.SHA }}
+        GH_TOKEN: ${{ secrets.NIGHTLY_TESTING }}
+        ZULIP_API_KEY: ${{ secrets.ZULIP_API_KEY }}
+      run: |
+        current_version="${NIGHTLY}"
+        bump_branch_suffix="${BUMP_BRANCH#bump/}"
```

Comment:
As usual, my suggestion would be to print this variable, for debugging purposes.